azcli_conformity.sh

```diff
@@ -0,0 +1,60 @@
+# Define Variables for the Configurations
+export APP_SECRET=$(uuidgen)
+
+# Ask User Input for the App Registration Name
+echo " "
+echo "Which name do you want to give to your App Registration?"
+IFS=$'\n'
+read -r APP_NAME
+echo " "
+echo "Hello, Thank you for trying this script out.
+We're configuring your Azure Account to work with Cloud One Conformity, this might take several minutes depending of how many subscriptions you have."
+
+# Create App Registration
+az ad app create --display-name "$APP_NAME" --password $APP_SECRET --credential-description secret --required-resource-accesses @manifest.json > /dev/null
```

Comment:
♻️ What about some new lines to make this a bit more readable?
```suggestion
az ad app create \
  --display-name "$APP_NAME" \
  --password $APP_SECRET \
  --credential-description secret \
  --required-resource-accesses @manifest.json > /dev/null
```

Same could be applied to the other long lines?

Reply:
Along with the review comments I have a few questions:

1. Agreed this could be handled better, there are more improvements I planned on making but due to my limited Git knowledge didn't want to make additional changes until we worked out what we start from, once we get this into a separate branch making feature branches based on that.
2. Planned on adding in an additional check for existing and update existing app rather than re-creating more
3. Ideally we have improved the logic before that so it gracefully handles multiple executions
4. Planned on adding argument support
5. Agree, also my preferred approach, arguments with all args include default values if not specified for ease of use for first time run, planned on adding arg support for selectively applying to subscriptions etc.
6. Will take a look at these haven't used this before.

Reply:
Great add!! definitely better to read this way.